app/src/views/campaigns/settings.vue

```diff
@@ -1,10 +1,217 @@
+          v-model="tempData.description"
+          :placeholder="$t('campaigns.placeholder.description')"
+        />
+      </el-form-item>
+      
+      <h2>Paramètres</h2>
+
+      <el-row :gutter="20">
+        <el-col :span="12">
+          <el-form-item
+            :label="$t('table.campaigns.isVisible')"
+            prop="maxProposition"
+          >
+            <el-switch
+              style="padding : 10px;"
+              v-model="tempData.isPublish"
+              active-color="#13ce66"
+              inactive-color="#ff4949"
+            />
+          </el-form-item>
+        </el-col>
+        <el-col :span="12">
+          <el-form-item
+            :label="$t('table.campaigns.date')"
+            prop="endAt"
+            :rules="[
+              { required: true, message: $t('form.campaigns.date.required') },
+              { validator: dateValidator, trigger: 'blur' },
+            ]"
+          >
+            <el-date-picker
+              v-model="tempData.endAt"
+              placeholder="Date de fin"
+              type="date"
+            />
+          </el-form-item>
+        </el-col>
+      </el-row>
+      <el-row :gutter="20">
+        <el-col :span="12">
+          <el-form-item
+            :label="$t('table.campaigns.maxProposition')"
+            prop="maxProposition"
+          >
+            <el-input-number
+              :min="0"
+              :max="50"
+              v-model="tempData.maxProposition"
+            />
+          </el-form-item>
+        </el-col>
+      </el-row>
+
+      <div class="dialog-footer">
+        <el-button type="danger" @click="archive">{{ $t('table.archive') }}</el-button>
+        <el-button @click="reset">{{ $t('table.reset') }}</el-button>
+        <el-button type="primary" @click="updateData">{{ $t('table.confirm') }}</el-button>
+      </div>
+    </el-form>
 
+    <progress-dialog ref="ProgressDialog" @close-after-success="reset" />
+  </div>
+</template>
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator';
 
-@Component({ name: 'CampaignsSettings' })
-export default class extends Vue {}
+import moment from 'moment';
+import { Component, Vue, Watch } from 'vue-property-decorator';
+import { cloneDeep } from 'lodash';
+import { Form } from 'element-ui';
+
+import { defaultCampaignData, createCampaign, deleteCampaign, getCampaign, updateCampaign } from '../../api/campaigns';
+import CategorySelect from '@/components/CategorySelect/index.vue';
+import ProgressDialog from '@/components/ProgressDialog/index.vue';
+import { CampaignsModule } from '../../store/modules/campaigns';
+
+
+@Component({ 
+  name: 'CampaignsSettings' ,
+  components: { ProgressDialog, CategorySelect },
+})
+
+export default class extends Vue {
+  private tempData: any = defaultCampaignData; 
+  
+  private cnt: number = 0;
+  private max: number = 1;
+ 
+  public get id() {
+    return Number(this.$route.params.id);
+  }
+
+  public created() {
+    this.$nextTick(() => {
```

Comment:
A comment about why you used $nextTick and link to documentation (vuejs) may be good

https://vuejs.org/v2/api/#Vue-nextTick

Reply:
```suggestion
    // Prevent error during component creation https://vuejs.org/v2/api/#Vue-nextTick
    this.$nextTick(() => {
```